### PR TITLE
New Article -1888235-2

### PR DESCRIPTION
**Pull request by rachana-satao**
New Article - How to ensure multiple controls bound to the same data source remain synchronized

This article is merging of - [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/multiple-controls-bound-to-data-source-synchronized?view=netframeworkdesktop-4.8), [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/how-to-apply-the-propertynamechanged-pattern?view=netframeworkdesktop-4.8), [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/change-notification-in-windows-forms-data-binding?view=netframeworkdesktop-4.8), [link](https://docs.microsoft.com/en-us/dotnet/desktop/winforms/how-to-implement-the-inotifypropertychanged-interface?view=netframeworkdesktop-4.8)

Preview - [link](https://review.docs.microsoft.com/en-us/dotnet/desktop/winforms/data/design-great-data-sources?view=netdesktop-5.0&branch=pr-en-us-1378)

### CODE REVIEW

File: dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+The example uses a <xref:System.Windows.Forms.BindingSource> to bind the data source and the controls. Alternatively, you can bind the controls directly to the data source and retrieve the <xref:System.Windows.Forms.BindingManagerBase> for the binding from the form's <xref:System.Windows.Forms.Control.BindingContext%2A> and then handle the <xref:System.Windows.Forms.BindingManagerBase.BindingComplete> event for the `BindingManagerBase`. For more information on bind the data source and the controls, see the help page about the `BindingComplete` event of `BindingManagerBase`.
+
+:::code language="csharp" source="snippets/multiple-controls-bound-to-data-source-synchronized/csharp/form1.cs" id="bind_controls_using_BindingSource":::
+
+:::code language="vb" source="snippets/multiple-controls-bound-to-data-source-synchronized/vb/form1.vb" id="bind_controls_using_BindingSource":::
+
+## How to apply the PropertyNameChanged pattern
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md:34]: Should there be a question mark at the end?

File: dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+:::code language="csharp" source="snippets/multiple-controls-bound-to-data-source-synchronized/csharp/form2.cs" id="apply_propertyNameChanged_pattern":::
+
+:::code language="vb" source="snippets/multiple-controls-bound-to-data-source-synchronized/vb/form2.vb" id="apply_propertyNameChanged_pattern":::
+
+## Change notification in Windows Forms data binding
+
+One of the most important concepts of Windows Forms data binding is the *change notification*. To ensure that your data source and bound controls always have the most recent data, you must add change notification for data binding. Specifically, you want to ensure that bound controls are notified of changes that were made to their data source. The data source is notified of changes that were made to the bound properties of a control.
+
+There are different kinds of change notification, depending on the kind of data binding:
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md:46]: There are different kinds of change notifications --> plural

File: dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### Change notification for custom controls
+
+Finally, from the control side you must expose a *PropertyName*Changed event for each property designed to be bound to data. The changes to the control property are then propagated to the bound data source. For more information, see [How to: Apply the PropertyNameChanged Pattern](/dotnet/desktop/winforms/how-to-apply-the-propertynamechanged-pattern?view=netframeworkdesktop-4.8&preserve-view=true)
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md:64]: space needed after *PropertyName*

File: dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+dev_langs: 
+  - "csharp"
+  - "vb"
+helpviewer_keywords: 
+  - "controls [Windows Forms], binding multiple"
+  - "controls [Windows Forms], synchronizing with data source"
+ms.custom: devdivchpfy22
+---
+
+# How to ensure multiple controls bound to the same data source remain synchronized
```
Comment (Lovina-Saldanha) [inline dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md:14]: Do we need a question mark here at the end?

File: dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md (hunk 3 above)
Comment (rachana-satao) [reply dotnet-desktop-guide/net/winforms/data/multiple-controls-bound-to-data-source-synchronized.md:64]: The method name format doesn't have space.

**Comment by rachana-satao** [comment]
@adegeo Acrolinx Score for this article - 100

**Comment by rachana-satao** [comment]
@adegeo As per discussion not adding example of first article in this article (it will be added to the fifth article).

**Review by adegeo — APPROVED** [review]

**Comment by adegeo** [comment]
In case this PR is looked at later, I did the following:

1. Change the title and add `Windows Forms .NET`
2. Moved the types of change notification to the top of the article considering 2/3 of the article is related to this and not the last 1/3
3. The last 1/3, sync multiple controls, is now last and points to the API reference page which has an example.

**Pull request merged**